### PR TITLE
Fix term parsers for PLC, UPLC, PIR. Now they can parse comments.

### PR DESCRIPTION
**Pull request by thealmarty**
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Changelog fragments have been written (if appropriate)
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [x] Targeting master unless this is a cherry-pick backport
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

**Comment by thealmarty** [comment]
Tested locally.

File: plutus-core/plutus-core/src/PlutusCore/Parser.hs [hunk 1]
```diff
@@ -69,7 +69,7 @@ errorTerm = inParens $ Error <$> wordPos "error" <*> pType
 
 -- | Parser for all PLC terms.
 term :: Parser PTerm
-term = choice $ map try
+term = whitespace >> (choice $ map try
```
Comment (effectfully) [inline plutus-core/plutus-core/src/PlutusCore/Parser.hs:72]: I'd write it as

```haskell
term = do
    whitespace
    choice $ map try
        <...>
```

but feel free to ignore my nitpicking.

**Review by effectfully — APPROVED** [review]
Yay!

File: plutus-core/plutus-core/src/PlutusCore/Parser.hs (hunk 1 above)
Comment (thealmarty) [reply plutus-core/plutus-core/src/PlutusCore/Parser.hs:72]: I like `do` in general too actually.

**Pull request merged**